netket/nn/symmetric_linear.py

```diff
@@ -101,62 +107,417 @@ def full_bias(self, bias):
         Convert symmetry-reduced bias of shape (features,) to the full bias of
         shape (n_symm * features,).
         """
-        return jnp.repeat(bias, self.n_symm)
+        return jnp.expand_dims(bias, (0, 2))
 
     @compact
-    def __call__(self, inputs: Array) -> Array:
+    def __call__(self, x: Array) -> Array:
         """Applies the symmetrized linear transformation to the inputs along the last dimension.
 
         Args:
-          inputs: The nd-array to be transformed.
+          x: The nd-array to be transformed.
 
         Returns:
           The transformed input.
         """
-        dtype = jnp.promote_types(inputs.dtype, self.dtype)
-        inputs = jnp.asarray(inputs, dtype)
+        dtype = jnp.promote_types(x.dtype, self.dtype)
+        x = jnp.asarray(x, dtype)
 
         kernel = self.param(
-            "kernel", self.kernel_init, (inputs.shape[-1], self.features), self.dtype
+            "kernel", self.kernel_init, (x.shape[-1], self.features), self.dtype
         )
-        kernel = self.full_kernel(kernel)
+
+        if self.mask:
+            kernel = kernel * jnp.expand_dims(self.mask, 1)
+
+        kernel = self.full_kernel(kernel).reshape(-1, self.features, self.n_symm)
         kernel = jnp.asarray(kernel, dtype)
 
-        y = lax.dot_general(
-            inputs,
+        x = lax.dot_general(
+            x,
             kernel,
-            (((inputs.ndim - 1,), (0,)), ((), ())),
+            (((x.ndim - 1,), (0,)), ((), ())),
             precision=self.precision,
         )
 
+        x = x.reshape(-1, self.features, self.n_symm)
```

Comment:
this changes the output of DenseSymm, right?

Before it used to return `[-1, features * n_symm]` instead now it will be `[-1, features, n_symm]`, is that right?

cc @femtobit would you be ok with this change?

Reply:
Yes, also sorry I forgot to mention this